mlflow/projects/backend/loader.py

```diff
@@ -0,0 +1,32 @@
+import entrypoints
+import logging
+
+ENTRYPOINT_GROUP_NAME = "mlflow.mlproject_backend"
+
+__logger__ = logging.getLogger(__name__)
+
+
+# Statically register backend defined in mlflow
+MLFLOW_BACKENDS = dict()
+
+
+def load_backend(backend_name):
+    # Static backends
+    if backend_name in MLFLOW_BACKENDS:
+        return MLFLOW_BACKENDS[backend_name]()
+
+    # backends from plugin
+    try:
+        backend_builder = entrypoints.get_single(ENTRYPOINT_GROUP_NAME,
```

Comment:
Nit: @jdlesage Can we resolve these entrypoints at module import time? (similar to what we do for the Tracking backend store registry here: https://github.com/mlflow/mlflow/blob/dea20035fc804654b552b520942eae5ed52f48d4/mlflow/tracking/_tracking_service/utils.py#L99-L102)

Reply:
I think it is better to keep them lazy and don't import the plugin if it is not needed. You don't control the plugin implementations, and some of them can contain bugs. If mlflow imports all existing plugins to register them and one of these plugins have a bug then mlflow will not start.
There is already this problem with other registries. If you install in your virtual env the dummy project located in `tests/resources`, there are these warnings when mlflow is imported:
```
/home/jd.lesage/workspace_python/mlflow/mlflow/tracking/_model_registry/utils.py:61: UserWarning: Failure attempting to register store for scheme "file-plugin": bad magic number in 'mlflow_test_plugin.sqlalchemy_store': b'\x03\xf3\r\n'
  _model_registry_store_registry.register_entrypoints()
```
To workaround this problem, currently, these exceptions are caught and transformed in warnings.  I think it is safer to raise exceptions (stop if the plugin is not correct) and load only necessary plugins. 

I created a week ago a issue to make all registries lazy: https://github.com/mlflow/mlflow/issues/2445

Reply:
@jdlesage Great point. In that case, I agree that we should stick with lazy entrypoint reading and move other store registries (e.g, Tracking) to this paradigm as well.